src/ts/UIFactory.ts

```diff
@@ -366,6 +367,16 @@ function adsUILayout() {
       new PlaybackToggleOverlay(),
       new AdStatusOverlay(),
       controlBar,
+      new TitleBar({
+        components: [
+          new Container({
+            components: [
+              new AdMessageLabel(),
+            ],
+            cssClasses: ['ui-titlebar-top'],
+          }),
+        ],
```

Comment:
Let's add `keepHiddenWithoutMetadata` to not show the shadow if there is no message. This way we overlay less of the ad which is better for ad-revenue.

```suggestion
        ],
        keepHiddenWithoutMetadata: true,
```

Reply:
Ah I see, thanks for the insight!